cumulusci/tasks/metadata_etl/tests/test_field_sets.py

```diff
@@ -0,0 +1,157 @@
+            <field>ContactId</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Subject</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Description</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>IncidentType__c</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>CreatedDate</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>ClosedDate</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Branch__c</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Severity__c</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <label>Incident Detail</label>
+    </fieldSets>
+</CustomObject>
+"""
+
+
+class TestAddFieldsToFieldSet:
+    def test_adds_fields(self):
+        task = create_task(
+            AddFieldsToFieldSet,
+            {
+                "field_set": "Case.IncidentDetail",
+                "fields": ["IncidentDateTime__c", "LocationNotes__c"],
+            },
+        )
+
+        # Validate that the fields have been added as displayedFields
+        tree = metadata_tree.fromstring(OBJECT_XML)
+        result = task._transform_entity(tree, "Case")
+        print(result.tostring())
+        fields = result.find("fieldSets", fullName="IncidentDetail").findall(
+            "displayedFields"
+        )
+        print(len(fields))
+
+        test_elem = next(f for f in fields if f.field.text == "IncidentDateTime__c")
+        assert test_elem is not None
+        assert test_elem.isFieldManaged.text == "false"
+        assert test_elem.isRequired.text == "false"
+
+        test_elem = next(f for f in fields if f.field.text == "LocationNotes__c")
+        assert test_elem is not None
+        assert test_elem.isFieldManaged.text == "false"
+        assert test_elem.isRequired.text == "false"
+
+        # Validate that the fields were removed from availableFields
+        fields = result.find("fieldSets", fullName="IncidentDetail").findall(
+            "availableFields"
+        )
+        assert "IncidentDateTime__c" not in (f.field.text for f in fields)
```

Comment:
_Contra_ the above, these are fine because there's no assumption about the length of the generator expression.